nixos/modules/programs/steam.nix

```diff
@@ -4,18 +4,44 @@ with lib;
 
 let
   cfg = config.programs.steam;
-in {
+  gamescopeCfg = config.programs.gamescope;
+
+  steam = pkgs.steam.override ({
+    extraLibraries = pkgs: with config.hardware.opengl;
+      if pkgs.hostPlatform.is64bit
+      then [ package ] ++ extraPackages
+      else [ package32 ] ++ extraPackages32;
+  } // optionalAttrs
+    (cfg.gamescopeSession.enable && gamescopeCfg.capSysNice)
+    {
+      buildFHSUserEnv = pkgs.buildFHSUserEnvBubblewrap.override {
+        # use the setuid wrapped bubblewrap
+        bubblewrap = "/run/wrappers";
+      };
+    }
+  );
+
+  gamescopeSessionFile = (pkgs.writeTextDir "share/wayland-sessions/steam.desktop" ''
+    [Desktop Entry]
+    Name=Steam
+    Comment=A digital distribution platform
+    Exec=${pkgs.writeShellScript "steam-gamescope" ''
```

Comment:
Can we expose this binary to systemPackages to have it in PATH for users without DMs?

Reply:
Ok, I did it here: https://github.com/PedroHLC/nixpkgs/commit/4e4c9644270ef21770c8d835a36da1c81b1c7bb1.

Opened TTY3, logged in, typed `steam-gamescope`, and this thing is working amazingly.

Reply:
I added this in the latest rebase.